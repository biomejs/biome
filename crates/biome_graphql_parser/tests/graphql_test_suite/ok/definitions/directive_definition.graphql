directive @example on FIELD

directive @example on FIELD_DEFINITION | ARGUMENT_DEFINITION

directive @example on
  | FIELD
  | FRAGMENT_SPREAD
  | INLINE_FRAGMENT

"deprecated" directive @delegateField(name: String!) repeatable on OBJECT | INTERFACE

directive @directive(directive: directive) on FIELD
