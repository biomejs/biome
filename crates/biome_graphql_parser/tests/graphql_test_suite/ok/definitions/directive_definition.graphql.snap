---
source: crates/biome_graphql_parser/tests/spec_test.rs
expression: snapshot
---
## Input
```graphql
directive @example on FIELD

directive @example on FIELD_DEFINITION | ARGUMENT_DEFINITION

directive @example on
  | FIELD
  | FRAGMENT_SPREAD
  | INLINE_FRAGMENT

"deprecated" directive @delegateField(name: String!) repeatable on OBJECT | INTERFACE

```

## AST

```
GraphqlRoot {
    bom_token: missing (optional),
    definitions: GraphqlDefinitionList [
        GraphqlDirectiveDefinition {
            description: missing (optional),
            directive_token: DIRECTIVE_KW@0..10 "directive" [] [Whitespace(" ")],
            at_token: AT@10..11 "@" [] [],
            name: GraphqlName {
                value_token: GRAPHQL_NAME@11..19 "example" [] [Whitespace(" ")],
            },
            arguments: missing (optional),
            repeatable_token: missing (optional),
            on_token: ON_KW@19..22 "on" [] [Whitespace(" ")],
            bitwise_or_token: missing (optional),
            locations: GraphqlDirectiveLocationList [
                GraphqlDirectiveLocation {
                    value_token: UPPER_FIELD_KW@22..27 "FIELD" [] [],
                },
            ],
        },
        GraphqlDirectiveDefinition {
            description: missing (optional),
            directive_token: DIRECTIVE_KW@27..39 "directive" [Newline("\n"), Newline("\n")] [Whitespace(" ")],
            at_token: AT@39..40 "@" [] [],
            name: GraphqlName {
                value_token: GRAPHQL_NAME@40..48 "example" [] [Whitespace(" ")],
            },
            arguments: missing (optional),
            repeatable_token: missing (optional),
            on_token: ON_KW@48..51 "on" [] [Whitespace(" ")],
            bitwise_or_token: missing (optional),
            locations: GraphqlDirectiveLocationList [
                GraphqlDirectiveLocation {
                    value_token: FIELD_DEFINITION_KW@51..68 "FIELD_DEFINITION" [] [Whitespace(" ")],
                },
                PIPE@68..70 "|" [] [Whitespace(" ")],
                GraphqlDirectiveLocation {
                    value_token: ARGUMENT_DEFINITION_KW@70..89 "ARGUMENT_DEFINITION" [] [],
                },
            ],
        },
        GraphqlDirectiveDefinition {
            description: missing (optional),
            directive_token: DIRECTIVE_KW@89..101 "directive" [Newline("\n"), Newline("\n")] [Whitespace(" ")],
            at_token: AT@101..102 "@" [] [],
            name: GraphqlName {
                value_token: GRAPHQL_NAME@102..110 "example" [] [Whitespace(" ")],
            },
            arguments: missing (optional),
            repeatable_token: missing (optional),
            on_token: ON_KW@110..112 "on" [] [],
            bitwise_or_token: PIPE@112..117 "|" [Newline("\n"), Whitespace("  ")] [Whitespace(" ")],
            locations: GraphqlDirectiveLocationList [
                GraphqlDirectiveLocation {
                    value_token: UPPER_FIELD_KW@117..122 "FIELD" [] [],
                },
                PIPE@122..127 "|" [Newline("\n"), Whitespace("  ")] [Whitespace(" ")],
                GraphqlDirectiveLocation {
                    value_token: FRAGMENT_SPREAD_KW@127..142 "FRAGMENT_SPREAD" [] [],
                },
                PIPE@142..147 "|" [Newline("\n"), Whitespace("  ")] [Whitespace(" ")],
                GraphqlDirectiveLocation {
                    value_token: INLINE_FRAGMENT_KW@147..162 "INLINE_FRAGMENT" [] [],
                },
            ],
        },
        GraphqlDirectiveDefinition {
            description: GraphqlDescription {
                graphql_string_value: GraphqlStringValue {
                    graphql_string_literal_token: GRAPHQL_STRING_LITERAL@162..177 "\"deprecated\"" [Newline("\n"), Newline("\n")] [Whitespace(" ")],
                },
            },
            directive_token: DIRECTIVE_KW@177..187 "directive" [] [Whitespace(" ")],
            at_token: AT@187..188 "@" [] [],
            name: GraphqlName {
                value_token: GRAPHQL_NAME@188..201 "delegateField" [] [],
            },
            arguments: GraphqlArgumentsDefinition {
                l_paren_token: L_PAREN@201..202 "(" [] [],
                arguments: GraphqlArgumentDefinitionList [
                    GraphqlInputValueDefinition {
                        description: missing (optional),
                        name: GraphqlName {
                            value_token: GRAPHQL_NAME@202..206 "name" [] [],
                        },
                        colon_token: COLON@206..208 ":" [] [Whitespace(" ")],
                        ty: GraphqlNonNullType {
                            base: GraphqlNamedType {
                                name: GraphqlName {
                                    value_token: GRAPHQL_NAME@208..214 "String" [] [],
                                },
                            },
                            excl_token: BANG@214..215 "!" [] [],
                        },
                        default: missing (optional),
                        directives: GraphqlDirectiveList [],
                    },
                ],
                r_paren_token: R_PAREN@215..217 ")" [] [Whitespace(" ")],
            },
            repeatable_token: REPEATABLE_KW@217..228 "repeatable" [] [Whitespace(" ")],
            on_token: ON_KW@228..231 "on" [] [Whitespace(" ")],
            bitwise_or_token: missing (optional),
            locations: GraphqlDirectiveLocationList [
                GraphqlDirectiveLocation {
                    value_token: UPPER_OBJECT_KW@231..238 "OBJECT" [] [Whitespace(" ")],
                },
                PIPE@238..240 "|" [] [Whitespace(" ")],
                GraphqlDirectiveLocation {
                    value_token: UPPER_INTERFACE_KW@240..249 "INTERFACE" [] [],
                },
            ],
        },
    ],
    eof_token: EOF@249..250 "" [Newline("\n")] [],
}
```

## CST

```
0: GRAPHQL_ROOT@0..250
  0: (empty)
  1: GRAPHQL_DEFINITION_LIST@0..249
    0: GRAPHQL_DIRECTIVE_DEFINITION@0..27
      0: (empty)
      1: DIRECTIVE_KW@0..10 "directive" [] [Whitespace(" ")]
      2: AT@10..11 "@" [] []
      3: GRAPHQL_NAME@11..19
        0: GRAPHQL_NAME@11..19 "example" [] [Whitespace(" ")]
      4: (empty)
      5: (empty)
      6: ON_KW@19..22 "on" [] [Whitespace(" ")]
      7: (empty)
      8: GRAPHQL_DIRECTIVE_LOCATION_LIST@22..27
        0: GRAPHQL_DIRECTIVE_LOCATION@22..27
          0: UPPER_FIELD_KW@22..27 "FIELD" [] []
    1: GRAPHQL_DIRECTIVE_DEFINITION@27..89
      0: (empty)
      1: DIRECTIVE_KW@27..39 "directive" [Newline("\n"), Newline("\n")] [Whitespace(" ")]
      2: AT@39..40 "@" [] []
      3: GRAPHQL_NAME@40..48
        0: GRAPHQL_NAME@40..48 "example" [] [Whitespace(" ")]
      4: (empty)
      5: (empty)
      6: ON_KW@48..51 "on" [] [Whitespace(" ")]
      7: (empty)
      8: GRAPHQL_DIRECTIVE_LOCATION_LIST@51..89
        0: GRAPHQL_DIRECTIVE_LOCATION@51..68
          0: FIELD_DEFINITION_KW@51..68 "FIELD_DEFINITION" [] [Whitespace(" ")]
        1: PIPE@68..70 "|" [] [Whitespace(" ")]
        2: GRAPHQL_DIRECTIVE_LOCATION@70..89
          0: ARGUMENT_DEFINITION_KW@70..89 "ARGUMENT_DEFINITION" [] []
    2: GRAPHQL_DIRECTIVE_DEFINITION@89..162
      0: (empty)
      1: DIRECTIVE_KW@89..101 "directive" [Newline("\n"), Newline("\n")] [Whitespace(" ")]
      2: AT@101..102 "@" [] []
      3: GRAPHQL_NAME@102..110
        0: GRAPHQL_NAME@102..110 "example" [] [Whitespace(" ")]
      4: (empty)
      5: (empty)
      6: ON_KW@110..112 "on" [] []
      7: PIPE@112..117 "|" [Newline("\n"), Whitespace("  ")] [Whitespace(" ")]
      8: GRAPHQL_DIRECTIVE_LOCATION_LIST@117..162
        0: GRAPHQL_DIRECTIVE_LOCATION@117..122
          0: UPPER_FIELD_KW@117..122 "FIELD" [] []
        1: PIPE@122..127 "|" [Newline("\n"), Whitespace("  ")] [Whitespace(" ")]
        2: GRAPHQL_DIRECTIVE_LOCATION@127..142
          0: FRAGMENT_SPREAD_KW@127..142 "FRAGMENT_SPREAD" [] []
        3: PIPE@142..147 "|" [Newline("\n"), Whitespace("  ")] [Whitespace(" ")]
        4: GRAPHQL_DIRECTIVE_LOCATION@147..162
          0: INLINE_FRAGMENT_KW@147..162 "INLINE_FRAGMENT" [] []
    3: GRAPHQL_DIRECTIVE_DEFINITION@162..249
      0: GRAPHQL_DESCRIPTION@162..177
        0: GRAPHQL_STRING_VALUE@162..177
          0: GRAPHQL_STRING_LITERAL@162..177 "\"deprecated\"" [Newline("\n"), Newline("\n")] [Whitespace(" ")]
      1: DIRECTIVE_KW@177..187 "directive" [] [Whitespace(" ")]
      2: AT@187..188 "@" [] []
      3: GRAPHQL_NAME@188..201
        0: GRAPHQL_NAME@188..201 "delegateField" [] []
      4: GRAPHQL_ARGUMENTS_DEFINITION@201..217
        0: L_PAREN@201..202 "(" [] []
        1: GRAPHQL_ARGUMENT_DEFINITION_LIST@202..215
          0: GRAPHQL_INPUT_VALUE_DEFINITION@202..215
            0: (empty)
            1: GRAPHQL_NAME@202..206
              0: GRAPHQL_NAME@202..206 "name" [] []
            2: COLON@206..208 ":" [] [Whitespace(" ")]
            3: GRAPHQL_NON_NULL_TYPE@208..215
              0: GRAPHQL_NAMED_TYPE@208..214
                0: GRAPHQL_NAME@208..214
                  0: GRAPHQL_NAME@208..214 "String" [] []
              1: BANG@214..215 "!" [] []
            4: (empty)
            5: GRAPHQL_DIRECTIVE_LIST@215..215
        2: R_PAREN@215..217 ")" [] [Whitespace(" ")]
      5: REPEATABLE_KW@217..228 "repeatable" [] [Whitespace(" ")]
      6: ON_KW@228..231 "on" [] [Whitespace(" ")]
      7: (empty)
      8: GRAPHQL_DIRECTIVE_LOCATION_LIST@231..249
        0: GRAPHQL_DIRECTIVE_LOCATION@231..238
          0: UPPER_OBJECT_KW@231..238 "OBJECT" [] [Whitespace(" ")]
        1: PIPE@238..240 "|" [] [Whitespace(" ")]
        2: GRAPHQL_DIRECTIVE_LOCATION@240..249
          0: UPPER_INTERFACE_KW@240..249 "INTERFACE" [] []
  2: EOF@249..250 "" [Newline("\n")] []

```
